modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/champion/PocketCastsChampionBottomSheetDialog.kt

```diff
@@ -0,0 +1,63 @@
+package au.com.shiftyjelly.pocketcasts.profile.champion
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import androidx.compose.ui.platform.ComposeView
+import androidx.core.view.doOnLayout
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsEvent
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTracker
+import au.com.shiftyjelly.pocketcasts.compose.AppTheme
+import au.com.shiftyjelly.pocketcasts.compose.CallOnce
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import au.com.shiftyjelly.pocketcasts.utils.rateUs
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class PocketCastsChampionBottomSheetDialog : BottomSheetDialogFragment() {
+
+    @Inject lateinit var theme: Theme
+
+    @Inject lateinit var analyticsTracker: AnalyticsTracker
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        val context = context ?: throw Exception("Context not found")
+        return ComposeView(context).apply {
+            setContent {
+                AppTheme(theme.activeTheme) {
+                    CallOnce {
+                        analyticsTracker.track(AnalyticsEvent.POCKET_CASTS_CHAMPION_DIALOG_SHOWN)
+                    }
+                    ChampionDialog(
+                        onRateClick = {
+                            analyticsTracker.track(AnalyticsEvent.POCKET_CASTS_CHAMPION_DIALOG_RATE_BUTTON_TAPPED)
+                            rateUs(context)
```

Comment:
Does it make sense to show an in-app rating dialog here?
If we agree to show it, changes need to be made to allow the dialog to be shown even if it was shown from the Stats screen earlier.

Reply:
Great question. I tried to make it simpler, but I raised a question about it here pdeCcb-67m-p2#comment-4847

Reply:
I've just confirmed that we are going to keep this as it is pdeCcb-67m-p2#comment-4850